### ISSUE TITLE
Gut reports runtime as 0.0s for long-running tests

### ISSUE THREAD

**Issue by nightblade9** [bitwes/Gut#309]
Background: I write a lot of roguelikes, and for procedural generation, the only way to check everything works is to write a long-running test that generates hundreds/thousands of things and validates that everything is as it should be.

I wrote a test that's slow, you can emulate it with a test to the effect of:

```
var start = OS.get_ticks_msec()
while (OS.get_ticks_msec() - start <= 10 * 1000): pass
```

This test will pass (or do whatever it is - generate/assert) for ten seconds. Gut reports the final runtime as 0.0s, for some reason. 

Even without the loop, when i was simply generating dungeons (200 iterations takes around 10s0, Gut reported the test time as 0.0s.

**Comment by bitwes** [comment]
TL;DR This is a bug.  Can you use `yield` instead of `while`?
```
yield(yield_for(10), YIELD)
```
https://github.com/bitwes/Gut/wiki/Yielding

This is probably due to the janky way Gut keeps track of time.  I added an elapsed time counter to the GUI a long time ago.  I then used that to output the time when the run is over.  Since it needs to display the time as it goes the value is updated in `_process`.  It should just display `now - start_time` instead of keeping track of it in a variable.  But since it stores it, it doesn't get updated because the `while` is blocking everything and another frame never happens and `_process` isn't called..  Using `yield` would work around this, but it's still a bug.

**Comment by bitwes** [comment]
I misread, per usual.  You aren't using a `while`, the generation code is just blocking the same way.  There's a chance that `call_deferred` and a `yield` would get this working the right way.  You could even `yield` to a signal from the generator-thing so you don't always have to wait for the full 10s.

Off the top of my head:
```
var level_generator = Generator.new()
call_deferred(level_generator, 'make level')
yield(yield_to(level_generator, 'done_generating', 10), YIELD)
```

**Comment by nightblade9** [comment]
I can't use call_deferred or yield, because the generation method returns all generated level data.

**Comment by nightblade9** [comment]
You're right about one thing: adding a `yield(get_tree(), "idle_frame"))` plays nicely with GUT. The UI doesn't freeze when I do this.

**Comment by nightblade9** [comment]
Yield is good enough for now. Sorry, I thought you meant "yield on creation." Turns out yielding on idle-frame fixes it, and doesn't drop performance much.
